fy22_ferry_study/attraction_travel_times/distance_direction_parser.py

```diff
@@ -0,0 +1,164 @@
+import googlemaps
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+from pathlib import Path
+import pandas as pd
+from shapely.geometry import MultiLineString
+import polyline
+import geopandas as gpd
+
+load_dotenv()
+api_key = os.getenv("google_api")
+gmaps = googlemaps.Client(key=api_key)
+GDRIVE_FOLDER = Path(os.getenv("GDRIVE_FOLDER"))
+origins = GDRIVE_FOLDER / "origin.csv"
+destinations = GDRIVE_FOLDER / "attractions.csv"
+origin_df = pd.read_csv(origins)
+destinations_df = pd.read_csv(destinations)
+now = datetime.now()
+
+
+origin = (
+    39.946196,
+    -75.139832,
+)  # hardcoded here since my origin doesn't change, but if yours does you can tweak the function to accept origins.
+
+
+def distance_duration_iteration(mode):
+    """returns a list of origin/destination dicts for driving or transit, respectively"""
+    driving_list = []
+    transit_list = []
+
+    def distance_duration(destination):
```

Comment:
functions-in-a-function aren't ideal, but there are some use cases where you could argue they make sense. I'd suggest breaking this out into its own top-level function.